### PR TITLE
Comparator for log summaries.

### PR DESCRIPTION
**Pull request by gmarek**
cc @mikedanese @davidopp @wojtek-t @fgrzadkowski

Assigning @mwielgus, as it'll be probably most convenient.

### CODE REVIEW

File: compare/src/compare-logs.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	fmt.Fprintf(w, ", %v: ", dataMax.node)
+	changeColorFloat64AndWrite(dataMax.generationRate, baselineMax.generationRate, allowedVariance, enableOutputColoring, w)
+	fmt.Fprint(w, ")")
+	return
+}
+
+func (d *ViolatingLogGenerationData) PrintToStdout(leftBuild, rightBuild int, enableOutputColoring bool) {
+	w := tabwriter.NewWriter(os.Stdout, 1, 0, 1, ' ', 0)
+	fmt.Fprint(w, "File\t")
+  printBuildNumber(leftBuild, w, enableOutputColoring)
```
Comment (mwielgus) [inline compare/src/compare-logs.go]: broken indent.

File: compare/src/compare-logs.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	fmt.Fprintf(w, "(%v: ", dataMin.node)
+	changeColorFloat64AndWrite(dataMin.generationRate, baselineMin.generationRate, allowedVariance, enableOutputColoring, w)
+	fmt.Fprintf(w, ", %v: ", dataMax.node)
+	changeColorFloat64AndWrite(dataMax.generationRate, baselineMax.generationRate, allowedVariance, enableOutputColoring, w)
+	fmt.Fprint(w, ")")
+	return
+}
+
+func (d *ViolatingLogGenerationData) PrintToStdout(leftBuild, rightBuild int, enableOutputColoring bool) {
+	w := tabwriter.NewWriter(os.Stdout, 1, 0, 1, ' ', 0)
```
Comment (mwielgus) [inline compare/src/compare-logs.go]: please, no single letter variables.

File: compare/src/compare-logs.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+		return result
+	}
+	leftAggregates := computeLogsAggregates(left)
+	rightAggregates := computeLogsAggregates(right)
+	if len(leftAggregates) == 0 || len(rightAggregates) == 0 {
+		glog.Warningf("No data found in at leaste one input\nleft: %v\nright: %v", leftAggregates, rightAggregates)
+		return result
+	}
+
+	for f := range leftAggregates {
```
Comment (mwielgus) [inline compare/src/compare-logs.go]: why f, no single letter variables please.

**Comment by mwielgus** [comment]
Fix build and expand single-letter variables into something more explicit - functions with 5 single-letter variables are hard to read.

**Comment by gmarek** [comment]
Ugh... I must have overwritten my commit fixing the build. Are k,v expressive enough, or those need to be expanded as well?

File: compare/src/compare-logs.go (hunk 3 above)
Comment (gmarek) [inline compare/src/compare-logs.go]: f is for 'file'

**Comment by mwielgus** [comment]
lgtm

**Comment by gmarek** [comment]
Flake: 
```
submit-queue_test.go:609: 15:Test16 state set to "success" but expected "pending"
```
cc @pmorie

**Pull request merged**